### ISSUE TITLE
Automatically indent ggplot2 code after dplyr pipe

### ISSUE THREAD

**Issue by eipi10** [rstudio/rstudio#2043]
This is an enhancement request to adjust the automatic code indenting feature so that it indents ggplot code that comes after a dplyr pipe. 

Example 1 below shows code for a basic ggplot plot. RStudio automatically indents each line after the first in an R script (or in a code chunk in an rmd document). However, when the ggplot code comes at the end of a dplyr pipe, as in Example 2, RStudio's automatic indenting keeps all of the ggplot code at the same indentation level, which I find harder to read. 

It would be nice to have the option of setting RStudio to automatically indent ggplot code, even if it comes after a pipe, as in Example 3 (which I've indented manually). I think this would make the code easier to read, since it would be more obvious where the data transformation pipe ends and the ggplot code begins, and would also format ggplot code the same way, regardless of whether it stands alone or comes after a pipe.

I realize I could put the data transformations inside the call to ggplot, as in Example 4, but for longer pipes, I find it easier to read the code if the data transformation chain is piped into ggplot. 

Obviously, this is isn't a major issue, but just a feature that would be nice to have available.

```
library(tidyverse)

# Example 1
ggplot(mtcars, aes(wt, mpg)) +
  geom_point() +
  scale_x_continuous(limits=c(0,20)) +
  labs(x="Label")

# Example 2
mtcars %>% 
  group_by(cyl) %>% 
  summarise(mpg=mean(mpg),
            wt=mean(wt)) %>% 
  ggplot(aes(wt, mpg)) +
  geom_point() +
  scale_x_continuous(limits=c(0,20)) +
  labs(x="Label")

# Example 3
mtcars %>% 
  group_by(cyl) %>% 
  summarise(mpg=mean(mpg),
            wt=mean(wt)) %>% 
  ggplot(aes(wt, mpg)) +
    geom_point() +
    scale_x_continuous(limits=c(0,20)) +
    labs(x="Label")

# Example 4
ggplot(mtcars %>% 
         group_by(cyl) %>% 
         summarise(mpg=mean(mpg),
                   wt=mean(wt)),
       aes(wt, mpg)) +
  geom_point() +
  scale_x_continuous(limits=c(0,20)) +
  labs(x="Label")
```

**Comment by ronblum** [comment]
@eipi10 Thank you for the suggestion! We'll consider it for future release of RStudio.

**Comment by jzadra** [comment]
And also please add the functionality to spread single line ggplot code connected by `+` into multi lines using "Code > Reformat Code"

**Comment by ronblum** [comment]
@jzadra This sound like a different, though related, request. Can you please open a separate issue for that? Thanks!

**Comment by eipi10** [comment]
Before this feature request gets automatically closed, I wanted to see if there's any chance it is going to get put on the "to do" list.

**Comment by eipi10** [comment]
@ronblum I just received the "marked as stale" notification and am checking in to see if there are any plans to add indenting for ggplot code that comes after a dplyr pipe.

**Comment by jzadra** [comment]
Yes, I don't want this to go stale as it would be a great addition to the code formatting!

**Comment by fisher-j** [comment]
Just want to confirm that we are having to manually indent each line of ggplot  code after a pipe.

**Comment by eipi10** [comment]
Just commenting to avoid automatic closure of this issue. Automatic indentation of ggplot code would still be a nice thing to have.
